launch_testing/launch_testing/__init__.py

```diff
@@ -53,14 +53,15 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
-        self.expected_output = b'\n'.join(self.expected_lines)
+        self.expected_output = b'\n'.join(self.expected_lines) + b'\n'
         self.left_over_stdout = b''
         self.left_over_stderr = b''
         self.stdout_data = io.BytesIO()
         self.stderr_data = io.BytesIO()
         self.regex_match = regex_match
         self.exit_on_match = exit_on_match
         self.matched = False
+        self.matched_exactly = False
```

Comment:
what I had in mind while writing this code was that `matched_exactly` is true if no unexpected lines were received, whereas `matched` can be true even if non-matching lines were found. so I think by the time `matched` is able to go true, `matched_exactly` will have had its chance to go true it it's supposed to. if `matched_exactly` isn't true by that time, it will never be true

e.g. if you are looking for 'a; b; c' and you get 'x; a; b; y; c; z' then I expect `matched_exactly` to be false and `matched` to be true. then it will stop processing. if 'a;b;c' is received again at a later stage, I wouldn't have expected that to be allowed to set `matched_exactly` true, because extra lines were still received in the meantime

does that match with what you would expect it to do?